cdap-api/src/main/java/co/cask/cdap/api/Admin.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api;
+
+import co.cask.cdap.api.dataset.DatasetManagementException;
+import co.cask.cdap.api.dataset.DatasetProperties;
+
+import javax.annotation.Nullable;
+
+/**
+ * This interface provides methods for operational calls from within a CDAP application.
+ */
+public interface Admin {
```

Comment:
@chtyim and I talked about that when I wrote the design. One issue with that is that we already have a DatasetAdmin as part of the dataset framework. We though about calling this DatasetClient, but we also already have one those... 

Are you thinking that we will have ProgramAdmin? LifecycleAdmin? ApplicationAdmin? and many more? And people would call getContext().getAdmin().getDatasetAdmin().<operation>? Another option is to define an interface DataAdmin and make Admin extend that. That combines the benefits of smaller dedicated interfaces with ease of use of calling getContext().getAdmin().operation.